pkg/controller/seed-controller-manager/mla/project_controller_test.go

```diff
@@ -0,0 +1,302 @@
+				},
+				{
+					name:     "create",
+					code:     409,
+					method:   "POST",
+					path:     "/api/orgs",
+					body:     `{"id":0,"name":"projectName-create","address":{"address1":"","address2":"","city":"","zipCode":"","state":"","country":""}}`,
+					response: `{"message": "name already taken"}`,
+				},
+			},
+			hasFinalizer: true,
+			err:          false,
+		},
+		{
+			name: "update org for project",
+			project: &kubermaticv1.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "create",
+					Annotations: map[string]string{grafanaOrgAnnotationKey: "1"},
+				},
+				Spec: kubermaticv1.ProjectSpec{
+					Name: "projectName",
+				},
+			},
+			requests: []request{
+				{
+					name:     "get org by id",
+					code:     200,
+					method:   "GET",
+					path:     "/api/orgs/1",
+					response: `{"id":1,"name":"projectName-oldname","address":{"address1":"","address2":"","city":"","zipCode":"","state":"","country":""}}`,
+				},
+				{
+					name:     "update",
+					code:     200,
+					method:   "PUT",
+					path:     "/api/orgs/1",
+					body:     `{"id":1,"name":"projectName-create","address":{"address1":"","address2":"","city":"","zipCode":"","state":"","country":""}}`,
+					response: `{"message": "name already taken"}`,
+				},
+			},
+			hasFinalizer: true,
+			err:          false,
+		},
+		{
+			name: "delete org for project",
+			project: &kubermaticv1.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "delete",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+				},
+				Spec: kubermaticv1.ProjectSpec{
+					Name: "projectName",
+				},
+			},
+			hasFinalizer: false,
+			requests: []request{
+				{
+					name:     "delete org by id",
+					code:     200,
+					method:   "DELETE",
+					path:     "/api/orgs/1",
+					response: `{}`,
+				},
+				{
+					name:     "get org by name",
+					code:     200,
+					method:   "GET",
+					path:     "/api/orgs/name/projectName-delete",
+					response: `{"id":1,"name":"projectName-delete","address":{"address1":"","address2":"","city":"","zipCode":"","state":"","country":""}}`,
+				},
+			},
+		},
+		{
+			name: "delete org for project, get failed",
+			project: &kubermaticv1.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "delete",
+					DeletionTimestamp: &metav1.Time{Time: time.Now()},
+					Finalizers:        []string{mlaFinalizer},
+				},
+				Spec: kubermaticv1.ProjectSpec{
+					Name: "projectName",
+				},
+			},
+			hasFinalizer: true,
+			err:          true,
+			requests: []request{
+				{
+					name:     "get org by name",
+					code:     404,
+					method:   "GET",
+					path:     "/api/orgs/name/projectName-delete",
+					response: `{"id":1,"name":"projectName-create","address":{"address1":"","address2":"","city":"","zipCode":"","state":"","country":""}}`,
+				},
+			},
+		},
```

Comment:
Should we actually fail if we're deleting the project, but the organization doesn't exist? In such cases, it would make sense just to do nothing.

Reply:
yeah, I was thinking about that,  but for that, I will need to add one more PR for grafanasdk, cause as for now, sdk doesn't differentiate `NotFound` error and other errors(like network connection). so we need to fix sdk first.